apis/v1alpha1/tlsroute_types.go

```diff
@@ -65,11 +65,29 @@ type TLSRouteStatus struct {
 
 // TLSRouteRule is the configuration for a given rule.
 type TLSRouteRule struct {
-	// Matches define conditions used for matching the rule against an
-	// incoming TLS handshake. Each match is independent, i.e. this
-	// rule will be matched if **any** one of the matches is satisfied.
-	// If unspecified, all requests from the associated gateway TLS
-	// listener will match.
+	// Matches define conditions used for matching the rule against incoming TLS
+	// connections. Each match is independent, i.e. this rule will be matched if
+	// **any** one of the matches is satisfied. If unspecified, all requests
+	// from the associated gateway listener will match.
+	//
+	// Each client request should map to a maximum of one route rule. If a
+	// request matches multiple rules, matching precedence MUST be determined in
+	// order of the following criteria, continuing on ties:
+	//
+	// * The longest matching SNI.
```

Comment:
Do we need to comment on how to compare wildcard strings?

Reply:
Good catch. I think "longest matching" would still be what we want even if wildcards were involved. We were missing guidance for what should be done if "*.example.com" and "b.example.com" both match - I added a new line item below this to clarify that precedence should be given to the longest precise match in the event of a tie.